test/foo/lib/config/test_configloader.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import Mock
+
+import pytest
+
+from foo.lib.config import ConfigLoader, ConfigLoaderException
+
+
+def reset_config_loader_singleton():
+    ConfigLoader._instance = None
+    ConfigLoader._instance_ready = False
+
+
+def test_load(monkeypatch):
+    reset_config_loader_singleton()
+    monkeypatch.setenv('VAR3', 'not bar')
+    cfg_loader = ConfigLoader(defaults={'VAR0': 'default value 0',
+                                        'VAR1': 'default value 1'})
+    config = cfg_loader.load('test/foo/lib/config/config.yaml')
+    assert config._config == {'VAR0': 'fizz',
+                              'VAR1': 'default value 1',
+                              'VAR2': 'foo',
+                              'VAR3': 'not bar'}
+    assert config.source == 'test/foo/lib/config/config.yaml'
+
+
+def test_load_missing():
```

Comment:
This test is for env == 'test' right ? 
If so i think it is good to rename this test and create same test for env == 'prod'  also.
Maybe as parametrized test.